pkgs/tools/graphics/argyllcms/default.nix

```
@@ -25,109 +101,61 @@ stdenv.mkDerivation rec {
 
-      # enable Printer device support
-      USE_PRINTER = false ;
+      # Remove bundled libraries, making the build fail if not provided externally
+      rm -rf tiff jpeg png usb zlib
 
-      # enable CMF Measurement device and accessory support (if present)
-      USE_CMFM = false ;
+      export AR="$AR rusc"
 
-      # Use ArgyllCMS version of libusb (deprecated - don't use)
-      USE_LIBUSB = false ;
-
-      # Compile in graph plotting code (Not fully implemented)
-      USE_PLOT = true ;		# [true]
-
-      JPEGLIB = ;
-      JPEGINC = ;
-      HAVE_JPEG = true ;
-
-      TIFFLIB = ;
-      TIFFINC = ;
-      HAVE_TIFF = true ;
-
-      PNGLIB = ;
-      PNGINC = ;
-      HAVE_PNG = true ;
-
-      ZLIB = ;
-      ZINC = ;
-      HAVE_Z = true ;
-
-      SSLLIB = ;
-      SSLINC = ;
-      HAVE_SSL = true ;
-
-      LINKFLAGS +=
-        ${lib.concatStringsSep " " (map (x: "-L${x}/lib") buildInputs)}
-        -lrt -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss
-        -ljpeg -ltiff -lpng -lssl ;
-    '';
-  in ''
-    cp ${jamTop} Jamtop
-    substituteInPlace Makefile --replace "-j 3" "-j $NIX_BUILD_CORES"
-    # Remove tiff, jpg and png to be sure the nixpkgs-provided ones are used
-    rm -rf tiff jpg png
-
-    export AR="$AR rusc"
+      cat "$extraJamtopPath" >> Jamtop
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    substituteInPlace Jambase --replace-fail "-m64" ""
   '';
 
-  buildInputs = [
-    libtiff libjpeg libpng libX11 libXxf86vm libXrandr libXinerama libXext
-    libXrender libXScrnSaver libXdmcp libXau openssl
-  ];
-
-  buildFlags = [ "all" ];
+  extraJamtop =
+    let
+      commonLinkFlags = "-ljpeg -lpng -ltiff -lusb-1.0 -lz -lssl";
+      markPresent = name: ''
+        HAVE_${name} = true ;
+        ${name}INC = ;
+        ${name}LIB = ;
+      '';
+    in
+    ''
+      ${markPresent "JPEG"}
+      ${markPresent "PNG"}
+      ${markPresent "SSL"}
+      ${markPresent "TIFF"}
+      ${markPresent "Z"}
+      ${markPresent "ZLIB"}
+
+      ${lib.optionalString stdenv.isLinux ''
+        LINKFLAGS = ${lib.concatMapStringsSep " " (dep: "-L${lib.getOutput "lib" dep}/lib") buildInputs} ;
+        LINKFLAGS += -ldl -lrt ${commonLinkFlags} ;
+        GUILINKFLAGS = -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss ;
+      ''}
+
+      ${lib.optionalString stdenv.isDarwin ''
+        LINKFLAGS += ${commonLinkFlags} ;
+      ''}
+    '';
 
-  makeFlags = [
-    "PREFIX=${placeholder "out"}"
-  ];
+  passAsFile = [ "extraJamtop" ];
 
   # Install udev rules, but remove lines that set up the udev-acl
   # stuff, since that is handled by udev's own rules (70-udev-acl.rules)
   postInstall = ''
-    rm -v $out/bin/License.txt
     mkdir -p $out/etc/udev/rules.d
-    sed -i '/udev-acl/d' usb/55-Argyll.rules
-    cp -v usb/55-Argyll.rules $out/etc/udev/rules.d/
+    install -m 0644 udev-rules/55-Argyll.rules $out/etc/udev/rules.d/
```

AFAIK adding the `-D` option creates the destination directory as needed, removing the need for `mkdir -p $out/etc/udev/rules.d` in the line above.
```suggestion
    install -Dm 0644 udev-rules/55-Argyll.rules $out/etc/udev/rules.d/
```

❓ Out of curiosity: Are udev-rules Linux specific? If yes installation could be omitted on darwin.

---

Applied